tests/reporter/kubernetes.go

```diff
@@ -1040,24 +1040,26 @@ func (r *KubernetesReporter) logClusterOverview() {
 	}
 }
 
-// getNodesWithVirtLauncher returns all node where a virt-launcher pod ran (finished) or still runs
-func getNodesWithVirtLauncher(virtCli kubecli.KubevirtClient) []string {
-	pods, err := virtCli.CoreV1().Pods(v1.NamespaceAll).List(context.Background(), metav1.ListOptions{LabelSelector: fmt.Sprintf(virtLauncherNameFmt, v12.AppLabel)})
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to fetch pods: %v\n", err)
-		return nil
-	}
+// getNodesRunningTests returns all node used by pods on test namespaces
+func getNodesRunningTests(virtCli kubecli.KubevirtClient) []string {
+	nodes := []string{}
+	for _, testNamespace := range testsuite.TestNamespaces {
+		pods, err := virtCli.CoreV1().Pods(testNamespace).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to fetch pods: %v\n", err)
+			return nil
+		}
 
-	nodeMap := map[string]struct{}{}
-	for _, pod := range pods.Items {
-		if pod.Spec.NodeName != "" {
-			nodeMap[pod.Spec.NodeName] = struct{}{}
+		nodeMap := map[string]struct{}{}
+		for _, pod := range pods.Items {
+			if pod.Spec.NodeName != "" {
+				nodeMap[pod.Spec.NodeName] = struct{}{}
+			}
 		}
-	}
 
-	nodes := []string{}
-	for k := range nodeMap {
-		nodes = append(nodes, k)
+		for k := range nodeMap {
+			nodes = append(nodes, k)
```

Comment:
`nodes` is not set so you can end up with duplicates. You probably want to make the `nodeMap` out of the scope of the `for` cycle and just transform it on return.

Reply:
Thanks, fixed.